### DISCUSSION TITLE
Names of environment variables used for multihost testing

### DISCUSSION THREAD

**Discussion by psss** [teemtee/tmt#2020] [Polls]
Each guest provided in the `provision` step config of a [multihost](https://tmt.readthedocs.io/en/latest/spec/plans.html#multihost) plan can specify a `role`. The plan is to provide the list of all guests assigned a given `role` to each test using an environment variable.

We need to decide how the `role` name should be mapped to the environment variable name as it may contain both lowercase and uppercase characters while, as for now, we keep all environment [variables](https://tmt.readthedocs.io/en/latest/overview.html#variables) consistent using uppercase only.

Let's decide this as soon as possible to prevent any backward-incompatible changes later. For the following config:

```yaml
provision:
    how: virtual
    name: replica-one
    role: rEpLiCa
```

There are three options:

1. keep unchanged
2. lowercase all chars
3. uppercase all chars

**Comment by kkaarreell** [comment]
Btw, I have verified that in Restraint the exact role is followed, i.e. `<task name="(keylime-tests) /Multihost/sync" role="CLIEnts">` results in a variable `CLIEnts` containing the respective hostname.

**Comment by lukaszachy** [comment]
Is 'role' case sensitive within tmt usage? e.g. If i have plan with `role: replica` and `role: REPLICA` will tmt treat as two distinct roles?

**Reply by happz** [reply]
~~The later, as the very basic comparison is used:~~

The `role`/`where` matching is case sensitive:

```python
    def enabled_on_guest(self, guest: 'Guest') -> bool:
        """ Check if the plugin is enabled on the specific guest """

        # FIXME: cast() - typeless "dispatcher" method
        where = cast(List[str], self.get('where'))

        if not where:
            return True

        return any(destination in (guest.name, guest.role) for destination in where)
```

**Reply by sopos** [reply]
It should be, I guess. If not, it should be fixed.

**Reply by lukaszachy** [reply]
So we have no choice but to keep role as-is for the variable name - otherwise more roles will report under same variable.

**Reply by psss** [reply]
I believe we still have a choice: Clearly state that it is not a good idea to have two separate roles with the same name, distinguished only by the case of the letter. We can do this by implementation: Provided role name could be always converted to lowercase.

**Reply by sopos** [reply]
Not being a good idea does not mean we should take this freedom out of the users.

**Comment by psss** [comment]
Summary from today's hacking session:

* Keep the `role` name intact, do not do any normalization
* Use the name exactly as provided by the user for the environment variable name
* In the very same way use it for the `TMT_GUEST_ROLE` value
* Warn user if the `provision` step contains two roles differing only in letter case

Thanks for your votes and feedback!
